### ISSUE TITLE
[NEW] Избыточное многоточие в выражениях

### ISSUE THREAD

**Issue by otymko** [1c-syntax/bsl-language-server#1139]
## Описание проблемы, ошибки, которую надо диагностировать

Плохая практика программирования - злоупотреблять использованием обращений через точку. Например для объектной модели при обращении через точку выполняется запрос к БД и получение всех полей. При чрезмерном использовании сказывается на производительности системы и читаемости кода.

**Плохо**:
```bsl
ФИО =  ДокументРеализации.Партнер.КонтактноеЛицо.Наименование;
```
**Хорошо**:
Получить данные заранее через запрос. Так же можно использовать методы из БСП по чтению реквизитов.

Еще пример:

**Плохо**:
```bsl
Элементы.Товары.ТекущиеДаннык.Реквизит = "Значение";
Элементы.Товары.ТекущиеДаннык.Реквизит2 = 100;
```
**Хорошо**:
```bsl
ТекущиеДанные = Элементы.Товары.ТекущиеДаннык;
...
ТекущиеДанные.Реквизит = "Значение";
ТекущиеДанные.Реквизит2 = 100;
```

### Тип [Статья на русском](https://1c-syntax.github.io/bsl-language-server/contributing/DiagnosticTypeAndSeverity)

* [ ] :ant: Ошибка
* [ ] :cop: Уязвимость
* [ ] :guardsman: Потенциальная уязвимость
* [x] :poop: Качество кода
* [ ] :trollface: Другое

### Важность [Статья на русском](https://1c-syntax.github.io/bsl-language-server/contributing/DiagnosticTypeAndSeverity)

* [ ] :broken_heart: Блокирующая / Blocker
* [ ] :heart: Критическая / Critical
* [x] :yellow_heart: Важная / Major
* [ ] :blue_heart: Незначительная / Minor
* [ ] :green_heart: Информационная / Info
* [ ] :revolving_hearts: Другое

### Тэги [Статья на русском](https://1c-syntax.github.io/bsl-language-server/contributing/DiagnosticTag)

* [ ] `STANDARD` - "Нарушение стандартов 1С"
* [ ] `LOCKINOS` - "Не будет работать в другой ОС"
* [ ] `SQL` - "Проблема с запросом"
* [x] `PERFORMANCE` - "Проблема производительности"
* [ ] `BRAINOVERLOAD` - "Непонятный код"
* [x] `BADPRACTICE` - "Плохая практика программирования"
* [ ] `CLUMSY` - "Излишние действия"
* [ ] `DESIGN` - "Ошибка в проектировании"
* [ ] `SUSPICIOUS` - "Подозрительный код"
* [ ] `UNPREDICTABLE` - "Непредсказуемо работающий код"
* [ ] `DEPRECATED` - "Устаревшая функциональность"
* [ ] `ERROR` - "Ошибочная конструкция"
* [ ] `LOCALIZE` - "Проблемы локализации"

### Время на исправление (минут)

5 минут

**Comment by asosnoviy** [comment]
100 FP

**Comment by qtLex** [comment]
Будут постоянно ругаться на
```Документ.ПоступлениеТоваров.СделатьКрасиво()``` 
```ОтборЭлемента = ЭлементУсловногоОформления.Отбор.Элементы.Добавить(Тип("ЭлементОтбораКомпоновкиДанных"))```


Критичность строчек
```Ссылка.ЕщеСсылка.ИЕщеСсылка.Значение```
и
```Элементы.Товары.ТекущиеДаннык.Реквизит = 100```
разные.

Первое потенциальная проблема производительности, а второе "вкусовщина" как мне кажется.

**Comment by Stepa86** [comment]
Вы свой код на джаве видели? На многоточие они решили ругаться.

Вот если б можно было тип выцепить и точно знать, что идет обращение к чему то ссылочному. Ну например из описания метода.

**Comment by nixel2007** [comment]
dfa

**Comment by otymko** [comment]
Да, тут нужен dfa. Но для начала можно ловить например 4-ре точки и больше.

**Comment by zeegin** [comment]
Мне кажется что много точек это не плохо. Мне нравятся флюент интерфейсы.
Плохо это когда идет неявное чтение из базы по ссылке.
Вот если ты видишь что в строке есть фрагмент `Ссылка.ЧтоТо` то надо ругаться. А в остальных случаях я бы не стал.
